plugins/auth-backend/src/lib/catalog/helpers.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  UserEntity,
+  serializeEntityRef,
+  RELATION_MEMBER_OF,
+} from '@backstage/catalog-model';
+import { TokenParams } from '../../identity';
+
+export function getEntityClaims(entity: UserEntity): TokenParams['claims'] {
+  const userRef = serializeEntityRef(entity);
+  if (typeof userRef !== 'string') {
+    throw new Error(
+      `Failed to serialize user entity ref, ${JSON.stringify(userRef)}`,
+    );
+  }
+
+  const membershipRefs =
+    entity.relations
+      ?.filter(r => r.type === RELATION_MEMBER_OF)
+      .map(r => {
+        const ref = serializeEntityRef(r.target);
+        if (typeof ref !== 'string') {
+          throw new Error(
+            `Failed to serialize relation entity ref, ${JSON.stringify(ref)}`,
+          );
+        }
+        return ref;
+      }) ?? [];
+
+  return {
+    sub: userRef,
```

Comment:
Yeah perhaps we should set this to the `id` instead, that'll make more sense. I think that no matter what auth data model we end up with we'll want a stable and unique ID for each user that can be used by plugins and other external services to store state tied to users.

Reply:
So, should we set `sub` to `entity.metadata.name` ? That's the `id` returned by `emailSignInResolver` as well.

Reply:
We talked about this a bit - let's make this the entity ref. If we want to, we can serialize it excluding the kind and namespace if they are the default (`User` and `default`) values. This would make the value still just "freben" for me, for example.

Reply:
Okay cool. Makes sense to leave this as the full user entity ref for now.